### DISCUSSION TITLE
Table row selection and scrolling

### DISCUSSION THREAD

**Discussion by elatedstar** [primefaces/community#770] [PrimeNG]
We are using PRIMENG diamond layout (^14.2.3), table (<p-table>) has following options:

resizableColumns="true"
scrollable="true"
virtualScroll="true"
virtualScrollItemSize="34"
scrollHeight="flex"
styleClass="p-datatable-striped p-datatable-sm"
 
Challenge: 
Requirement is to scroll the table row-by-row using the arrow keys.  While scrolling each row should be selected to propagate event for selected row, further we also need to control the speed of the scroll. We couldn’t find built-in functionality in primeng table for row selection and control scrolling speed.
 
Workaround (Not working well):
We had to manually intercept and prevent default the key event, and throw in a debounce time to dynamically select the next row and use focus() on the html element to force the table to scroll. The issue is that we cannot get the table to scroll smoothly and keep the focused row at the bottom of the table (while scrolling down, top of table while scrolling up). The focused row seems to jump to the middle of the table now and then.

**Comment by cetincakiroglu** [comment]
Hi,

Could you please try it with the latest version of the diamond template (v17)?

**Comment by cetincakiroglu** [comment]
Closing the discussion due to inactivity.
